tomviz/DataSource.cxx

```diff
@@ -79,23 +78,15 @@ class DataSource::DSInternals
       vtkAlgorithm::SafeDownCast(this->ProducerProxy->GetClientSideObject());
     Q_ASSERT(alg);
     auto data = alg->GetOutputDataObject(0);
-
     auto fd = data->GetFieldData();
-    if (!this->TiltAngles) {
+    if (!fd->HasArray("tilt_angles")) {
       int* extent = vtkImageData::SafeDownCast(data)->GetExtent();
       int numTiltAngles = extent[5] - extent[4] + 1;
       vtkNew<vtkDoubleArray> array;
       array->SetName("tilt_angles");
       array->SetNumberOfTuples(numTiltAngles);
       array->FillComponent(0, 0.0);
-      if (!fd->HasArray("tilt_angles")) {
-        fd->AddArray(array.GetPointer());
-      }
-      this->TiltAngles = array.Get();
-    } else {
-      if (!fd->HasArray("tilt_angles")) {
-        fd->AddArray(this->TiltAngles);
-      }
+      fd->AddArray(array.GetPointer());
```

Comment:
You don't need GetPointer anymore, vtkNew was updated to do the auto conversion like vtkSmartPointer.

Reply:
Ok, will update